ansible/roles/build-grsec-kernel/tasks/compile.yml

```diff
@@ -1,37 +1,10 @@
 ---
-- name: Clean source directory prior to extraction.
-  file:
-    path: "{{ linux_source_directory }}"
-    state: absent
-
-- name: Extract Linux tarball (.tar -> directory).
-  unarchive:
-    copy: no
-    src: "{{ download_directory }}/linux-{{ linux_kernel_version }}.tar"
-    dest: "{{ download_directory }}"
-
-- name: Apply grsecurity patch.
-  shell: patch -p1 < ../{{ grsecurity_patch.filename | quote }}
-  args:
-    chdir: "{{ linux_source_directory }}"
-
-- name: Copy kernel config file.
-  copy:
-    src: config-{{ grsecurity_build_strategy }}
-    dest: "{{ linux_source_directory }}/.config"
-
-- name: Clean kernel source tree.
-  command: make-kpkg clean
-  args:
-    chdir: "{{ linux_source_directory }}"
-
-- name: Generate config, accepting default choices.
-  shell: yes '' | make-kpkg configure
-  args:
-    chdir: "{{ linux_source_directory }}"
-
 - name: Build the grsecurity-patched kernel.
-  command: fakeroot make-kpkg --initrd kernel_image
+  command: >
+    fakeroot make-kpkg
+    --revision 10.00.{{ grsecurity_build_strategy }}
```

Comment:
When building for multiple platforms, the `fetch` task would clobber previously built debs due to the filename being identical. Inserting the build strategy into the filename makes it clear which packages should be installed on which hosts.